server/sonar-server/src/main/java/org/sonar/server/authentication/UserIdentityAuthenticator.java

```diff
@@ -76,21 +77,30 @@ private UserDto register(UserIdentity user, IdentityProvider provider) {
         registerExistingUser(dbSession, userDto, user, provider);
         return userDto;
       }
-      return registerNewUser(dbSession, user, provider);
+      return registerNewUser(dbSession, user, provider, source);
     } finally {
       dbClient.closeSession(dbSession);
     }
   }
 
-  private UserDto registerNewUser(DbSession dbSession, UserIdentity user, IdentityProvider provider) {
+  private UserDto registerNewUser(DbSession dbSession, UserIdentity user, IdentityProvider provider, AuthenticationEvent.Source source) {
     if (!provider.allowsUsersToSignUp()) {
-      throw new UnauthorizedException(format("'%s' users are not allowed to sign up", provider.getKey()));
+      throw AuthenticationException.newBuilder()
+        .setSource(source)
+        .setLogin(user.getLogin())
+        .setMessage(format("'%s' users are not allowed to sign up", provider.getKey()))
+        .build();
     }
 
     String email = user.getEmail();
     if (email != null && dbClient.userDao().doesEmailExist(dbSession, email)) {
-      throw new UnauthorizedException(format(
-        "You can't sign up because email '%s' is already used by an existing user. This means that you probably already registered with another account.", email));
+      throw AuthenticationException.newBuilder()
```

Comment:
When doing some functional test, something is now no more working => these errors are not generating stacktrace in server and the message is no more displayed in the ui.
This is because UnauthorizedException was used for functional error, but as now AuthenticationException is used, such exceptions are displayed in server.